dex/order/order.go

```diff
@@ -752,6 +752,16 @@ func ValidateOrder(ord Order, status OrderStatus, lotSize uint64) error {
 	return nil
 }
 
+// ExtractAddress extracts the address from the order. If the order is a cancel
+// order, an empty string is returned.
+func ExtractAddress(ord Order) string {
+	trade := ord.Trade()
+	if trade == nil {
+		return ""
+	}
+	return trade.Address
+}
+
```

Comment:
It makes more sense to me for this (and ValidateOrder actually) to be part of the interface. I guess it's fine though.